test/role.test.js

```diff
@@ -511,6 +546,24 @@ describe('role model', function() {
       });
     });
 
+    it('should resolve user by id using custom user principalType', function(done) {
+      ACL.resolvePrincipal('User', user.id, function(err, u) {
+        if (err) return done(err);
+
+        expect(u.id).to.eql(user.id);
+
+        done();
+      });
+    });
+
+    it('should throw error when principalType is incorrect', function(done) {
+      ACL.resolvePrincipal('incorrectPrincipalType', user.id, function(err, u) {
+        assert(err);
+
+        done();
+      });
+    });
+
```

Comment:
testing here if ACL.resolvePrincipal throws an error if principalType is neither USER, ROLE, APPLICATION, nor permit to resolve a user model